js/run_unit_tests.ts

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env deno --allow-run
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+// This test is executed as part of tools/test.py
+// But it can also be run manually: ./target/debug/deno js/run_unit_tests.ts
+// You must have ./tools/http_server.py running in the background if you run
+// this manually.
+
+import { args, exit, run } from "deno";
+
+function extractNumber(pattern: RegExp, str: string) {
+  const matches = pattern.exec(str);
+  if (matches == null || pattern.exec(str) != null) {
+    return;
+  }
+  return Number(matches[1]);
+}
+
+function parseUnitTestOutput(
+  output: string
+): [string | undefined, string | undefined] {
```

Comment:
1. this function should not allow to return undefined for either 'expected' or 'actual'.
2. rather than returning a tuple, return an object `{ actual: number, expected: number }`.